modulus/models/module.py

```diff
@@ -65,55 +106,286 @@ def debug(self):
+                f"Model checkpoint {directory.joinpath('model.json')} not found"
+            )
 
-        file_name = Path(file_name)
-        if not file_name.parents[0].is_dir():
+        if not directory.joinpath("metadata.json").exists():
             raise IOError(
-                f"Model checkpoint parent directory {file_name.parents[0]} not found"
+                f"Model checkpoint {directory.joinpath('metadata.json')} not found"
             )
 
-        torch.save(self.state_dict(), str(file_name))
+        if not directory.joinpath("model.pt").exists():
+            raise IOError(
+                f"Model checkpoint {directory.joinpath('model.pt')} not found"
+            )
 
-    def load(self, file_name: Union[str, None] = None) -> None:
-        """Simple utility for loading the model from checkpoint
+        # Check if the checkpoint version is compatible with the current version
+        with open(directory.joinpath("metadata.json"), "r") as f:
+            metadata_info = json.load(f)
+            if metadata_info['mdlus_file_version'] != Module.__version__:
+                raise IOError(
+                    f"Model checkpoint version {metadata_info['mdlus_file_version']} is not compatible with current version {Module.__version__}"
+                )
+
+        return directory
+
+    def load(self, file_name: str) -> None:
+        """Simple utility for loading the model weights from checkpoint
 
         Parameters
         ----------
-        file_name : Union[str,None], optional
-            Checkpoint file name. When none is provide it will default to the model's
-            name set in the meta data, by default None
+        file_name : str
+            Checkpoint file name
 
         Raises
         ------
         IOError
-            If file_name provided does not exist
+            If file_name provided does not exist or is not a valid checkpoint
         """
-        if file_name is None:
-            file_name = self.meta.name + ".pt"
 
-        file_name = Path(file_name)
-        if not file_name.exists():
-            raise IOError(f"Model checkpoint {file_name} not found")
+        directory = Module._check_checkpoint(file_name)
 
-        model_dict = torch.load(file_name, map_location=self.device)
+        model_dict = torch.load(
+            directory.joinpath("model.pt"), map_location=self.device
+        )
         self.load_state_dict(model_dict)
 
+    @classmethod
+    def from_checkpoint(cls, file_name: str) -> None:
+        """Simple utility for constructing a model from a checkpoint
+
+        Parameters
+        ----------
+        file_name : str
+            Checkpoint file name
+
+        Returns
+        -------
+        Module
+
+        Raises
+        ------
+        IOError
+            If file_name provided does not exist or is not a valid checkpoint
+        """
+
+        directory = Module._check_checkpoint(file_name)
+
+        with open(directory.joinpath("args.json"), "r") as f:
```

Comment:
Like in the earlier comment, would be especially useful here to use a filesystem abstraction so we can support both local as well as remote filesystems or NGC registries down the line